### PR TITLE
Add towncrier

### PR DESCRIPTION
**Pull request by AndrewFerr**
Signed-off-by: Andrew Ferrazzutti <andrewf@element.io>

This is required for compliance with our [contributing guide](https://github.com/matrix-org/matrix-appservice-bridge/blob/develop/CONTRIBUTING.md).

### CODE REVIEW

**Review by Half-Shot — APPROVED** [review]
This looks fine but it would be nice to know why CI is failing for changelog?

**Comment by AndrewFerr** [comment]
> why CI is failing for changelog?

Because it came before https://github.com/matrix-org/matrix-appservice-discord/pull/790 :slightly_smiling_face:  Once that's merged, I'll rebase this change onto it.  Not only should CI be green, but CI should test this towncrier setup too!

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+[tool.towncrier]
```
Comment (Half-Shot) [inline pyproject.toml:1]: N.B. `.gitkeep` -> `git.keep`

Comment (AndrewFerr) [reply pyproject.toml:1]: Still no good...any ideas?
```
ERROR: newsfragment changelog.d/git.keep does not end with a '.' or '!'
!! Please see the contributing guide for help writing your changelog entry:
https://github.com/matrix-org/matrix-appservice-irc/blob/develop/CONTRIBUTING.md#making-pull-requests
Error: Process completed with exit code 1.
```

File: scripts/check-newsfragment [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+# make sure that origin/develop is up to date
+git remote set-branches --add origin develop
+git fetch -q origin develop
+
+pr="$PULL_REQUEST_NUMBER"
+
+# Print a link to the contributing guide if the user makes a mistake
+CONTRIBUTING_GUIDE_TEXT="!! Please see the contributing guide for help writing your changelog entry:
+https://github.com/matrix-org/matrix-appservice-irc/blob/develop/CONTRIBUTING.md#making-pull-requests"
```
Comment (Half-Shot) [inline scripts/check-newsfragment]: This needs updating

File: scripts/check-newsfragment [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+# If check-newsfragment returns a non-zero exit code, print the contributing guide and exit
+python3 -m towncrier.check --compare-with=origin/develop || (echo -e "$CONTRIBUTING_GUIDE_TEXT" >&2 && exit 1)
+
+echo
+echo "--------------------------"
+echo
+
+matched=0
+for f in $(git diff --diff-filter=d --name-only FETCH_HEAD... -- changelog.d); do
+    # check that any added newsfiles on this branch end with a full stop.
```
Comment (Half-Shot) [inline scripts/check-newsfragment:30]: I suspect we need something in here to filter out `git.keep`?

File: scripts/check-newsfragment (hunk 2 above)
Comment (AndrewFerr) [reply scripts/check-newsfragment]: This was copied from [matrix-appservice-bridge](https://github.com/matrix-org/matrix-appservice-bridge/blob/develop/scripts/check-newsfragment#L19), so it needs updating too...!

**Comment by AndrewFerr** [comment]
This should be good to go.

We should also consider sharing some of these settings across all projects, like the towncrier version, the newsfile check script, etc.  ATM each project does it slightly differently and it's a bit of a headache to context-switch between them.

**Review by Half-Shot — APPROVED** [review]

**Pull request merged**